lib/overcommit/hook/pre_commit/foodcritic.rb

```diff
@@ -0,0 +1,95 @@
+module Overcommit::Hook::PreCommit
+  # Runs `foodcritic` against any modified Ruby files from chef directory structure.
+  #
+  # @see http://www.foodcritic.io/
+  #
+  # Provides three configuration options:
+  #
+  # * `cookbooks_directory`
+  #   If not set, or set to false, the current directory will be treated as a cookbook directory
+  #   (the one containing recipes, libraries, etc.)
+  #   If set to path (absolute or relative), hook will interpret it as a cookbooks directory
+  #   (so all each subdirectory will be treated as separate cookbook)
+  # * `environments_directory`
+  #   If provided, the given path will be treated as environments directory
+  # * `roles_directory`
+  #   If provided, the given path will be treated as roles directory
+  #
+  # By default, none of those options is set, which means, the repo directory will be treaded
+  # as a cookbook directory (with recipes, libraries etc.)
+  #
+  # Example:
+  #
+  # Foodcritic:
+  #   enabled: true
+  #   cookbooks_directory: './cookbooks'
+  #   environments_directory: './environments'
+  #   roles_directory: './roles'
+  class Foodcritic < Base
+    def run
+      args = modified_environments_args + modified_roles_args + modified_cookbooks_args
+
+      args += applicable_files.reject do |file|
+        %w[spec test].any? { |dir| file.include?("#{File::SEPARATOR}#{dir}#{File::SEPARATOR}") }
```

Comment:
Yeah, I included that because I had a problem with `foodcritic` interpreting them as a separate cookbooks, and I couldn't resolve it easily. It appears, that `-B root_path` which you did was the solution I was looking for :)